app/src/main/java/de/tum/in/tumcampusapp/component/other/settings/SettingsFragment.kt

```diff
@@ -192,6 +193,11 @@ class SettingsFragment : PreferenceFragmentCompat(), Preference.OnPreferenceClic
                 StartSyncReceiver.cancelBackground()
             }
         }
+
+        // restart app after language change
+        if (key == "language_preference") {
+            if (activity != null) (activity as SettingsActivity).restartApp()
```

Comment:
Please merge the two conditionals into one and && them.

Reply:
Done.